### PR TITLE
Implement structures

### PR DESCRIPTION
**Pull request by Schaeff**
Implement structures in ZoKrates:
```
from "./bar.code" import Bar

struct Foo {
   Bar bar
   field[3] baz
}

def main(private Foo f) -> (Bar):
   return f.bar
```

## Changes
- New explicit syntax for imports: `from PATH import SYMBOL as ALIAS`, we have backward compatibility by aliasing `import PATH` to `from PATH import main as PATH.FILENAME`
- structs can be declared, defined, accessed, mutated

### CODE REVIEW

File: t.code [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from "./u.code" import Fooo
+
+struct Bar {
+	a: field,
+	a: field,
```
Comment (JacobEberhardt) [inline t.code]: Should be 'b'

File: zokrates_core/src/typed_absy/variable.rs [hunk 2]
```diff
@@ -27,6 +26,10 @@ impl<'ast> Variable<'ast> {
         Self::with_id_and_type(id, Type::array(ty, size))
     }
 
+    pub fn struc(id: Identifier<'ast>, ty: Vec<(MemberId, Type)>) -> Variable<'ast> {
```
Comment (stefandeml) [inline zokrates_core/src/typed_absy/variable.rs]:
```suggestion
    pub fn struct(id: Identifier<'ast>, ty: Vec<(MemberId, Type)>) -> Variable<'ast> {
```

File: zokrates_core/src/static_analysis/propagation.rs [hunk 3]
```diff
@@ -302,10 +327,124 @@ impl<'ast, T: Field> Folder<'ast, T> for Propagator<'ast, T> {
...
+    fn fold_struct_expression_inner(
+        &mut self,
+        ty: &Vec<(MemberId, Type)>,
+        e: StructExpressionInner<'ast, T>,
+    ) -> StructExpressionInner<'ast, T> {
+        match e {
+            StructExpressionInner::Identifier(id) => {
+                match self
+                    .constants
+                    .get(&TypedAssignee::Identifier(Variable::struc(
```
Comment (stefandeml) [inline zokrates_core/src/static_analysis/propagation.rs]:
```suggestion
                    .get(&TypedAssignee::Identifier(Variable::struct(
```

File: zokrates_core/src/static_analysis/constrain_inputs.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
+//! Add runtime boolean checks on user inputs
```
Comment (stefandeml) [inline zokrates_core/src/static_analysis/constrain_inputs.rs]: That means we now check for all Boolean inputs strictly.

Eg in that example code:
```go
struct A {
        bool b
}

def main(A a) -> (field):
  field dummy = 12
  return dummy
```

we will add boolean checks for the input `a`.

File: zokrates_core/src/static_analysis/propagation.rs (hunk 3 above)
Comment (Schaeff) [reply zokrates_core/src/static_analysis/propagation.rs]: `struct` is reserved in Rust, cannot be used for a function name

File: zokrates_core/src/typed_absy/variable.rs (hunk 2 above)
Comment (Schaeff) [reply zokrates_core/src/typed_absy/variable.rs]: struct is a keyword

**Pull request merged**